src/lib.rs

```diff
@@ -83,7 +83,7 @@ impl EyreHandler for Handler {
             write!(f, "\n\nCaused by:")?;
 
             let multiple = cause.source().is_some();
-            let errors = std::iter::successors(Some(cause), |e| e.source());
+            let errors = std::iter::successors(Some(cause), |e| e.clone().source());
```

Comment:
This should work and hopefully will be more obvious about what it's doing. Because yea, I'm not really sure why clone fixes this either...

```suggestion
            let errors = std::iter::successors(Some(cause), |e| (*e).source());
```

Reply:
I figured out what is happening, this is a very fun trait implementation :D 

https://doc.rust-lang.org/stable/src/core/clone.rs.html#222-227